go/vt/sqlparser/ast.go

```diff
@@ -5393,14 +5393,19 @@ type GroupConcatExpr struct {
 	Distinct  string
 	Exprs     SelectExprs
 	OrderBy   OrderBy
-	Separator string
+	Separator Separator
+}
+
+type Separator struct {
+	SeparatorString  string
+	DefaultSeparator bool
 }
 
 // Format formats the node
 func (node *GroupConcatExpr) Format(buf *TrackedBuffer) {
-	sep := node.Separator
-	if sep != "" {
-		sep = " separator " + "'" + sep + "'"
+	sep := ""
+	if !node.Separator.DefaultSeparator {
+		sep = " separator " + "'" + node.Separator.SeparatorString + "'"
```

Comment:
It would be good to add some testing to this PR, too. I think we could test this logic reasonably well by using this `Format` function to output the parsed version of the statement. 

Check out the `TestValid` test function and the valid test cases it runs. I think some statements that use group_concat and show that we can include or not include the empty string and get the correct results would be a good way to add coverage for this change.